api/server/server.go

```diff
@@ -402,6 +610,18 @@ func (s *Server) startGears(ctx context.Context, cancel context.CancelFunc) {
 
 	installChildReaper()
 
+	if s.nodeType == ServerTypePureRunner {
+		// Run grpc too
+		grpcAddr := fmt.Sprintf("%s:%d", whoAmI().String(), s.grpcListenPort)
+		pr, err := agent.CreatePureRunner(grpcAddr, s.agent, s.cert, s.certKey, s.certAuthority)
+		if err != nil {
+			logrus.WithError(err).Fatal("Pure runner server creation error")
+		}
+		go func() {
+			pr.Start()
```

Comment:
For the shutdown case, (interrupt + sigterm), we should log the error returned from Start(). See the go-routine a few lines below. There's also a Serve & Listen error case handling in there (see cancel()).

Reply:
https://github.com/fnproject/fn/pull/814/files#diff-44a527ea72ea9572e1cf71c7f1e2a0f0R632

Reply:
GracefulStop case needs handling here.. See:
https://github.com/fnproject/fn/pull/814/files#diff-44a527ea72ea9572e1cf71c7f1e2a0f0R645

Reply:
We added a call to cancel() in case of error in creating the pure runner or error in start it.